src/main/java/com/pileproject/drivecommand/model/nxt/NxtMachineSimulator.java

```diff
@@ -0,0 +1,52 @@
+package com.pileproject.drivecommand.model.nxt;
+
+import com.pileproject.drivecommand.machine.device.DeviceType;
+import com.pileproject.drivecommand.machine.device.input.LineSensor;
+import com.pileproject.drivecommand.machine.device.input.SoundSensor;
+import com.pileproject.drivecommand.machine.device.input.TouchSensor;
+import com.pileproject.drivecommand.machine.device.input.simulation.SimulatedLineSensor;
+import com.pileproject.drivecommand.machine.device.input.simulation.SimulatedSoundSensor;
+import com.pileproject.drivecommand.machine.device.input.simulation.SimulatedTouchSensor;
+import com.pileproject.drivecommand.machine.device.output.Motor;
+import com.pileproject.drivecommand.machine.device.output.simulation.SimulatedMotor;
+import com.pileproject.drivecommand.machine.device.port.InputPort;
+import com.pileproject.drivecommand.machine.device.port.OutputPort;
+
+public class NxtMachineSimulator extends NxtMachine {
```

Comment:
Your opinion seems right.
One thing which made me confused was that NxtController takes NxtMachine as its constructor argument, which seems to have to be MachineBase.
And now I feel that the controller's name is misleading as it implies general enough to be able to control any nxt machines.